api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -106,6 +107,34 @@ public User getUserByUsername(String username) {
 		return users.get(0);
 	}
 	
+	
+	/**
+	 * @see org.openmrs.api.UserService#getUserByEmail(java.lang.String)
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public User getUserByEmail(String email) {
+		return (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.eq("email", email).ignoreCase()).uniqueResult();	
+	}
+	
+	/**
+	 * @see org.openmrs.api.UserService#getUserByActivationKey(java.lang.String)
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public User getUserByActivationKey(String activationKey) {
+		User ret = null;
+		User user = (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.like("activationKey", activationKey, MatchMode.START)).uniqueResult();	
+		if(user != null) {
+			String[] tokens = user.getActivationKey().split(":");
+			if(tokens[0].equals(activationKey)) {
+				ret = user;
```

Comment:
It's cleaner to replace this with **return user;** remove line 126 and replace line134 with **return null;**